pkg/integrations/windows_exporter/config.go

```diff
@@ -0,0 +1,132 @@
+package windows_exporter //nolint:golint
+
+import (
+	"reflect"
+
+	"github.com/go-kit/kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+func init() {
+	integrations.RegisterIntegration(&Config{})
+}
+
+// Config controls the windows_exporter integration.
+// All of these and their child fields are pointers so we can determine if the value was set or not.
+type Config struct {
+	Common config.Common `yaml:",inline"`
+
+	EnabledCollectors string `yaml:"enabled_collectors"`
+
+	Exchange    *ExchangeConfig    `yaml:"exchange"`
+	IIS         *IISConfig         `yaml:"iis"`
+	TextFile    *TextFileConfig    `yaml:"text_file"`
+	SMTP        *SMTPConfig        `yaml:"smtp"`
+	Service     *ServiceConfig     `yaml:"service"`
+	Process     *ProcessConfig     `yaml:"process"`
+	Network     *NetworkConfig     `yaml:"network"`
+	MSSQL       *MSSQLConfig       `yaml:"mssql"`
+	MSMQ        *MSMQConfig        `yaml:"msmq"`
+	LogicalDisk *LogicalDiskConfig `yaml:"logical_disk"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config.
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+func (c *Config) Name() string {
+	return "windows_exporter"
+}
+
+func (c *Config) CommonConfig() config.Common {
+	return c.Common
+}
+
+func (c *Config) NewIntegration(l log.Logger) (integrations.Integration, error) {
+	return New(l, c)
+}
+
+// The Windows Collector takes a map of configuration to set, so we need to convert from agent config to a key value
+// using the windows_exporter key name 'collector.iis.site-whitelist' for example.
+func (c *Config) ConvertToMap() map[string]string {
+	configMap := make(map[string]string)
+	mapToConfig(c, configMap)
+	return configMap
+}
+
+type ExchangeConfig struct {
+	EnabledList *string `yaml:"enabled_list" exporter:"collectors.exchange.enabled"`
```

Comment:
Do these need to be pointers?

Reply:
I think so. The issue arises that passing in "" would overwrite the defaults that windows_exporter uses. If there is another way to do it then would def like that route.